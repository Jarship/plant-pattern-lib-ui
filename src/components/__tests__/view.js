import React from "react";
import renderer from "react-test-renderer";
import { matchers } from 'jest-emotion';

import View from "../view";

expect.extend(matchers);

const mockTheme = {
  header: {
    height: 0
  },
  breakpoints: ["768px"],
  space: base => base,
  views: {
    primary: {
      backgroundColor: "red",
    }
  }
};
mockTheme.views.primary.padding = mockTheme.space(4);

test("view conforms to expected styles", () => {
  const testView = renderer.create(<View theme={mockTheme} />).toJSON();
  const {
    views: { primary: { backgroundColor }},
    header: { height },
    breakpoints: [mobile],
    space
  } = mockTheme;

  expect(testView).toHaveStyleRule("background-color", `${backgroundColor}`);
  expect(testView).toHaveStyleRule("padding", `${space(4)}px`);
  expect(testView).toHaveStyleRule("min-width", "100vw");
  expect(testView).toHaveStyleRule("overflow-x", "hidden");
  expect(testView).toMatchSnapshot();
});
