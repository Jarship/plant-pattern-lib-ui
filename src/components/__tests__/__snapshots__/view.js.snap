// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view conforms to expected styles 1`] = `
<article
  className="css-1rc4b0p-lib--ui-view"
/>
`;
